app/forms/provider_interface/offer_wizard.rb

```diff
@@ -38,21 +44,68 @@ def valid_for_current_step?
       valid?(current_step.to_sym)
     end
 
-    def next_step
-      index = STEPS[decision.to_sym].index(current_step.to_sym)
+    def next_step(step = current_step)
+      index = STEPS[decision.to_sym].index(step.to_sym)
+      return unless index
 
-      STEPS[decision.to_sym][index + 1] if index
+      current_step = STEPS[decision.to_sym][index + 1]
+
+      return save_and_go_to_next_step(current_step) if current_step.eql?(:providers) && available_providers.one?
+      return save_and_go_to_next_step(current_step) if current_step.eql?(:courses) && available_courses.one?
+      return save_and_go_to_next_step(current_step) if current_step.eql?(:study_modes) && available_study_modes.one?
+      return save_and_go_to_next_step(current_step) if current_step.eql?(:locations) && available_course_options.one?
+
+      current_step
     end
 
+    delegate :previous_step, to: :wizard_path_history
+
   private
 
+    def save_and_go_to_next_step(step)
```

Comment:
As per the thread above we renamed the variable in order to avoid confusion with current step. We don’t want to change anything because we are not saving current step, we dont want to persist it either.

Reply:
I think my confusion is because the next_step method has this side effect where it sets some state on the wizard. It's a bit unexpected, but also minor